### PR TITLE
Improve space v2

### PR DESCRIPTION
**Pull request by txsmith**
Fixed:
- Panning
- Zooming
- Space resize
- Moving groups out of view
- Use SVG to draw links
- Create links at user defined positions
- Styling

Todo:
- [x] Fix tests
- [ ] Show arrow heads
- [ ] Fix some styling issues when zoom is applied
- [ ] Make linking two groups twice impossible

### CODE REVIEW

File: src/projects/pull-request/space/vector-behavior.html [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    },
+
+    clipBox: function(pos, size, northWest, southEast) {
+      var clippedTopLeft = this.clip(pos, northWest, southEast);
+      var bottomRight = this.add(clippedTopLeft, size);
+      var clippedBottomRight = this.clip(bottomRight, northWest, southEast);
+      return this.subtract(clippedBottomRight, size);
+    },
+
+    between: function(v1, v2) {
```
Comment (TimvdLippe) [inline src/projects/pull-request/space/vector-behavior.html]: I think average sounds better

File: src/projects/pull-request/space/vector-behavior.html [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      y: f(v.y)
+    };
+  };
+
+  var reduce = function(f) {
+    return function() {
+      var args = Array.prototype.slice.call(arguments);
+      return args.reduce(function(a, b) {
+        return joinAll(a, b, f);
+      }.bind(this));
```
Comment (TimvdLippe) [inline src/projects/pull-request/space/vector-behavior.html]: This bind is not required

**Comment by eanker** [comment]
It seems to work.
There are some cases (especially with the borders) that the links behave in a strange way and are decoupled.

**Comment by TimvdLippe** [comment]
2 minor issues extracted, other than that LGTM!

**Pull request merged**